Prevent initialize from being called multiple times

copperwall opened the pull request:
This prevents the google analytics tag from being loaded more than once when `ReactGA.initialize` is called.

Now if client code calls `ReactGA.initialize('UA-fasdf')` multiple times, the GA tag will only be added once, and a `create` impression will only be sent once.

Closes #399 
Closes #267

antondomratchev commented:
+1. I was about to open an issue with the problem that this PR solves. Would love to see this merged in for the next release.

JClackett commented:
Please merge this! :)

copperwall commented:
Oops, I didn't realize this broke all of the tests 😬 I'll see what I can do about those

antondomratchev commented:
@copperwall I believe most of the breakages are because each test is not cleaning up. I was thinking of adding an `afterEach` that sets `window.ga = null` which might only work in cases where init was called. Most likely this will require reworking some tests to rely on a `beforeEach` setup call that would init the tracker.
just my 2c

bigblobby commented:
Any progress on this?

copperwall commented:
sorry this kind of fell off my plate, I can take a stab at the tests again soon

SimeonC commented on package.json:
```diff
@@ -89,8 +89,8 @@
     "react-addons-test-utils": "^15.0.1",
     "react-dom": "^15.0.1",
-    "react-router": "4.4.0",
-    "react-router-dom": "4.4.0",
```
Not sure why this is on a beta version over the non rc version?

SimeonC replied:
Eh, since when did they un-publish versions 😨

SimeonC commented on src/utils/loadGA.js:
```diff
@@ -1,11 +1,19 @@
+
+function loadGA(options) {
+  const gaAddress = optionToGaAddress(options);
+
```
I don't think we need a memoize function, we only support one `ga` being loaded so just doing a boolean here would be fine.
```
let isLoaded = false;
function loadGA(options) {
  if (isLoaded) return;
  ...
```

copperwall replied:
Ah thanks, that makes this PR a lot less complicated. It seems to play a lot more nicely with the test suite as well.

copperwall commented:
@SimeonC Thank you for the review! I can make some changes this weekend

SimeonC commented:
@copperwall sorry this has a bunch of conflicts now 😅 

Good news is that whenever you can get this updated it'll auto-release without me procrastinating for weeks 😆 

Hopefully, this should also encourage me to actually review things faster as I can do it all on github now 🚀

copperwall commented:
@SimeonC sorry it took so long to get these changes in. I fixed the merge conflicts and dialed back the memoization change and added the boolean change like you suggested.

If there's anything else this needs I'm happy to help!

SimeonC commented:
Looks good to me, thanks!

copperwall commented:
Thanks for the quick response!
